collection-scripts/gather_ctlplane_resources

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# load shared functions and data
+DIR_NAME=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+source "${DIR_NAME}/common.sh"
+
+NS="$1"
+if [ -z "$NS" ]; then
+    echo "No namespace passed, using the default one"
+    NS=openstack
+fi
+
+# Only get resources if the namespace exists
+if ! check_namespace "${NS}"; then
+    exit 0
+fi
+
+declare resources=(
+    "services"
+    "routes"
+    "daemonset"
+    "replicaset"
+    "deployments"
+    "statefulset"
+    "jobs"
+)
+
+# Get pods
+for p in $(oc -n "$NS" get pods -o custom-columns=":metadata.name"); do
+    echo "Dump logs for pod: $p";
+    mkdir -p "${NAMESPACE_PATH}"/"$NS"/pods/"$p"/logs
+    # describe pod
+    /usr/bin/oc -n "$NS" describe pod "$p" > "${NAMESPACE_PATH}"/"$NS"/pods/"$p"/"$p"-describe.log
+    # get logs
+    /usr/bin/oc -n "$NS" logs --prefix=true --all-containers=true "$p" > "${NAMESPACE_PATH}"/"$NS"/pods/"$p"/logs/"$p".log;
+    # dump --previous logs for all the terminated containers of the pod
+    cur=$(/usr/bin/oc -n "$NS" get pods "$p"  -o jsonpath="{.status.containerStatuses[*].lastState.terminated}")
+    if [[ -n "$cur" ]]; then
+        /usr/bin/oc -n "$NS" logs --prefix=true --previous "$p" > "${NAMESPACE_PATH}"/"$NS"/pods/"$p"/logs/"$p"-previous.log;
```

Comment:
@stuggi I guess this is enough to get logs from the previous container (if it has been terminated), right?
/cc @Sandeepyadav93

Reply:
I tried it against a ctlplane where glance is in `CrashLoopBackOff` state:

```
[must-gather-7jfqh] OUT namespaces/openstack/pods/glance-external-api-56696597c8-99m4h/logs/glance-external-api-56696597c8-99m4h-previous.log
[must-gather-7jfqh] OUT namespaces/openstack/pods/glance-external-api-56696597c8-99m4h/logs/glance-external-api-56696597c8-99m4h.log
[must-gather-7jfqh] OUT namespaces/openstack/pods/glance-internal-api-689db5d84b-7zq6q/
[must-gather-7jfqh] OUT namespaces/openstack/pods/glance-internal-api-689db5d84b-7zq6q/glance-internal-api-689db5d84b-7zq6q-describe.log
[must-gather-7jfqh] OUT namespaces/openstack/pods/glance-internal-api-689db5d84b-7zq6q/logs/
[must-gather-7jfqh] OUT namespaces/openstack/pods/glance-internal-api-689db5d84b-7zq6q/logs/glance-internal-api-689db5d84b-7zq6q-previous.log
[must-gather-7jfqh] OUT namespaces/openstack/pods/glance-internal-api-689db5d84b-7zq6q/logs/glance-internal-api-689db5d84b-7zq6q.log
```

Reply:
yes, I have just seen it after I commented on the doc. should have checked the PR, sorry